glue/plugins/wcs_autolinking/wcs_autolinking.py

```diff
@@ -60,43 +86,144 @@ def __init__(self, data1=None, data2=None, cids1=None, cids2=None):
-            # WCS, which will only work if the celestial coordinates are separable.
-            # TODO: find a more generalized APE 14-compatible way to do this.
+            # A generalized APE 14-compatible way
+            # Handle also the extra-spatial axes such as those of the time and wavelength dimensions
+
+            wcs1_celestial_axis_physical_types = None
+            wcs2_celestial_axis_physical_types = None
 
             if not wcs1.has_celestial or not wcs2.has_celestial:
-                raise IncompatibleWCS("Can't create WCS link between {0} and {1}".format(data1.label, data2.label))
+                slicing_axes_celestial1 = []
+                slicing_axes_celestial2 = []
+            elif wcs1.has_celestial and wcs2.has_celestial:
+                wcs1_celestial_axis_physical_types = wcs1.celestial.world_axis_physical_types
+                wcs2_celestial_axis_physical_types = wcs2.celestial.world_axis_physical_types
+
+                cids1 = data1.pixel_component_ids
+                cids2 = data2.pixel_component_ids
+                cids1_celestial = [cids1[wcs1.wcs.naxis - wcs1.wcs.lng - 1],
+                                   cids1[wcs1.wcs.naxis - wcs1.wcs.lat - 1]]
+                cids2_celestial = [cids2[wcs2.wcs.naxis - wcs2.wcs.lng - 1],
+                                   cids2[wcs2.wcs.naxis - wcs2.wcs.lat - 1]]
+
+                slicing_axes_celestial1 = [cids1_celestial[0].axis, cids1_celestial[1].axis]
+                slicing_axes_celestial1 = sorted(slicing_axes_celestial1, key=str, reverse=False)
+
+                slicing_axes_celestial2 = [cids2_celestial[0].axis, cids2_celestial[1].axis]
+                slicing_axes_celestial2 = sorted(slicing_axes_celestial2, key=str, reverse=False)
+            elif wcs1.has_celestial and not wcs2.has_celestial:
+                wcs1_celestial_axis_physical_types = wcs1.celestial.world_axis_physical_types
+                cids1 = data1.pixel_component_ids
+                cids1_celestial = [cids1[wcs1.wcs.naxis - wcs1.wcs.lng - 1],
+                                   cids1[wcs1.wcs.naxis - wcs1.wcs.lat - 1]]
+                slicing_axes_celestial1 = [cids1_celestial[0].axis, cids1_celestial[1].axis]
+                slicing_axes_celestial1 = sorted(slicing_axes_celestial1, key=str, reverse=False)
+
+                slicing_axes_celestial2 = []
+
+            elif not wcs1.has_celestial and wcs2.has_celestial:
+                slicing_axes_celestial1 = []
+
+                wcs2_celestial_axis_physical_types = wcs2.celestial.world_axis_physical_types
+                cids2 = data2.pixel_component_ids
+                cids2_celestial = [cids2[wcs2.wcs.naxis - wcs2.wcs.lng - 1],
+                                   cids2[wcs2.wcs.naxis - wcs2.wcs.lat - 1]]
+                slicing_axes_celestial2 = [cids2_celestial[0].axis, cids2_celestial[1].axis]
+                slicing_axes_celestial2 = sorted(slicing_axes_celestial2, key=str, reverse=False)
```

Comment:
If you know one WCS doesn't have celestial axes, is there any need to keep track of the celestial axes for the WCS that does? This will then get ignored later, right? In other words, if one WCS doesn't have celestial axes, why not set both ``slicing_axes_celestial1`` and ``slicing_axes_celestial2`` to ``[]``?

Reply:
Yup, agreed.